Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -483,16 +490,23 @@ final class iCloudSynchronizer {
 	/**
 	Determine which data source has the latest data available by comparing the timestamps of the local and remote sources.
 	*/
-	private func latestDataSource(forKey key: Defaults.Keys) -> Defaults.iCloud.DataSource {
+	private func latestDataSource(forKey key: Defaults.Keys) -> Defaults.iCloud.DataSource? {
+		let remoteTimestamp = timestamp(forKey: key, source: .remote)
+		let localTimestamp = timestamp(forKey: key, source: .local)
+
+		return switch (remoteTimestamp, localTimestamp) {
+		// If the local timestamp does not exist, use the remote timestamp as the latest data source.
+		case (.some(_), nil):
+			.remote
 		// If the remote timestamp does not exist, use the local timestamp as the latest data source.
-		guard let remoteTimestamp = timestamp(forKey: key, source: .remote) else {
-			return .local
+		case (nil, .some(_)):
+			.local
+		case let (.some(localTimestamp), .some(remoteTimestamp)):
```

Comment:
This is inverted.

Make sure to add a test for this case that would have failed.

Reply:
My fault, thought it was already covered by the tests.
Added a test for `latestDataSource`.